### PR TITLE
fix(eslint-plugin): delete [] in message if ReadOnly<string[]> is detected

### LINKED ISSUE #9498: Bug: ["@typescript-eslint/array-type"] suggests incorrect type when specifying "default"
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVJxwB4BReEADwxFWAGd4AlEKYPCATyLoxi1QDmAbQC6APjEAKSrygAueCQCUCkgG4AsAChtQA&eslintrc=N4KABBYEQA4DYFcDmBLAdgZygLjAbXEgigAEAXATxgFMMBjAJxRjIFpa50ypCIBdADS9oDBHFo4woIsXJVajZmw5cA9AEMGDdRVaUakgjMhRqWgPYMoxosKJCbU6ABNqAM3VjuuKJu26MFABbeGprAF87MD5hSPCgA&tsconfig=&tokens=false

### Repro Code

```TypeScript
declare function foo<E extends Readonly<string[]>>(extra: E): E;
```

### ESLint Config

```javascript
{
    "plugins": [
      "@typescript-eslint"
    ],
    "rules": {
      "@typescript-eslint/array-type": [
        "error, { "default": "array-simple" }
      ]
    }
}
```

### tsconfig

_No response_

### Expected Result

Expected error message: `Array type using 'Readonly<string[]>' is forbidden for simple types. Use 'readonly string[]' instead.`

### Actual Result

The actual error message is incorrect--`Array type using 'Readonly<string[]>' is forbidden for simple types. Use 'readonly string[][]' instead. 1:32 - 1:50` It suggests a type that is not the same as the written type.

### Additional Info

I encountered these while looking though the CI tests for DefinitelyTyped. All of the cases have the same extra `[]` in error message regardless of the type used (not just `string`-- the wrong message also happens with generic types, type parameters, and constant/defined types). 

Removing the `default` part of the config and just specifying `"@typescript-eslint/array-type": [ "error" ]` gives the expected error message.

### PR DESCRIPTION
**Pull request by YutongZhuu**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9498
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

I made string manipulations in the context.report to delete the extra [], I am not sure if this is the idea way of fixing the issue permanently.  Tests are failing, I believe they are failing on the main branch too, not caused by my changes.

### CODE REVIEW

**Comment by abrahamguo** [comment]
Were you able to find out why this bug was happening/what was the root cause?

**Comment by YutongZhuu** [comment]
> Were you able to find out why this bug was happening/what was the root cause?
```       
 const messageId =
          currentOption === 'array'
            ? isReadonlyWithGenericArrayType
              ? 'errorStringArrayReadonly'
              : 'errorStringArray'
            : 'errorStringArraySimple';
```

Here is the logic for determining which message to display. It has a currentOption of arraySimple, so the error will be ``        "Array type using '{{className}}<{{type}}>' is forbidden for simple types. Use '{{readonlyPrefix}}{{type}}[]' instead.",
``. And the ``getMessageType(type)`` function call returns string[]. That is why we have two []'s.

**Comment by abrahamguo** [comment]
Looking at how `errorStringArray` and `errorStringArrayReadonly` are defined, it may be better to be consistent with that, and add a new message `errorStringArraySimpleReadonly`.

Also, the tests are no longer failing on `main`, so you can pull from there.

**Comment by YutongZhuu** [comment]
> Looking at how `errorStringArray` and `errorStringArrayReadonly` are defined, it may be better to be consistent with that, and add a new message `errorStringArraySimpleReadonly`.
> 
> Also, the tests are no longer failing on `main`, so you can pull from there.

Makes sense, I update my PR soon.

File: packages/eslint-plugin/src/rules/array-type.ts
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/array-type.ts:1]: [Testing] We can't take any bugfixes to lint rules that don't have corresponding unit tests. Could you please add to `array-type.test.ts`?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for starting on this (and it was nice to see conversations around it) - but it'll need to have tests to be reviewed. Could you please? 🙂

File: packages/eslint-plugin/src/rules/array-type.ts
Comment (YutongZhuu) [reply packages/eslint-plugin/src/rules/array-type.ts:1]: Sure, I will update the PR.

Comment (YutongZhuu) [reply packages/eslint-plugin/src/rules/array-type.ts:1]: Hi, I've added a test case for this scenario, pls review.

**Comment by YutongZhuu** [comment]
> Thanks for starting on this (and it was nice to see conversations around it) - but it'll need to have tests to be reviewed. Could you please? 🙂

@JoshuaKGoldberg Hi, I have added a test as requested, I see this PR has not been reviewed yet, does it mean there is something else I need to do, or you guys simply do not have time yet?

**Comment by auvred** [comment]
As per [our contributing guidelines](https://typescript-eslint.io/contributing/pull-requests#addressing-feedback-and-beyond) this PR is in the queue of PRs to review, and will be reviewed when we are able.

This project is run by volunteer maintainers, so it might be a bit before we can find the time to get to it.

**Comment by YutongZhuu** [comment]
> As per [our contributing guidelines](https://typescript-eslint.io/contributing/pull-requests#addressing-feedback-and-beyond) this PR is in the queue of PRs to review, and will be reviewed when we are able.
> 
> 
> 
> This project is run by volunteer maintainers, so it might be a bit before we can find the time to get to it.

No, problem. I am just making sure I am not doing anything that prevents you from reviewing it.

**Review by JoshuaKGoldberg — APPROVED** [review]
👍 LGTM, thanks!

Mr Bean giving a weirdly enthusiastic thumbs up

**Comment by YutongZhuu** [comment]
> CI / Lint with build (lint) (pull_request)
is not successful, something I need to fix?

**Comment by JoshuaKGoldberg** [comment]
Oh yeah don't worry about that, #9794.

**Pull request merged**